src/utils.ts

```diff
@@ -232,6 +232,87 @@ const mergeObjects = <T extends object>(...objects: T[]): T => {
 const moveElementToEnd = <T>(array: T[], index: number): T[] =>
     array.slice(0, index).concat(array.slice(index + 1), array[index]);
 
+const queryStringParser = (url: string, keys: string[]): Dictionary<string> => {
+    let urlParams;
+    let results = {};
+
+    if (!url) return results;
+
+    if (typeof URL !== 'undefined' && typeof URLSearchParams !== 'undefined') {
+        const urlObject = new URL(url);
+        urlParams = new URLSearchParams(urlObject.search);
+    } else {
+        urlParams = queryStringParserFallback(url);
+    }
+
+    keys.forEach(key => {
+        const value = urlParams.get(key);
+        if (value) {
+            results[key] = value;
+        }
+    });
+
+    return results;
+};
+
+interface URLSearchParamsFallback {
+    get: (key: string) => string | null;
+}
+
+const queryStringParserFallback = (url: string): URLSearchParamsFallback => {
+    var params = {};
+    var queryString = url.split('?')[1];
+    var pairs = queryString.split('&');
+
+    pairs.forEach(pair => {
+        var [key, value] = pair.split('=');
+        if (key && value) {
+            params[key] = decodeURIComponent(value || '');
+        }
+    });
+
+    return {
+        get: function(key: string) {
+            return params[key];
+        },
+    };
+};
+
+// Get cookies as a dictionary
+const getCookies = (keys?: string[]): Dictionary<string> => {
+    // Helper function to parse cookies from document.cookie
+    const parseCookies = (): string[] => {
+        if (typeof window === 'undefined') {
+            return [];
+        }
+        return window.document.cookie.split(';').map(cookie => cookie.trim());
+    };
+
+    // Helper function to filter cookies by keys
+    const filterCookies = (cookies: string[], keys?: string[]): Dictionary<string> => {
+        const results: Dictionary<string> = {};
+    for (const cookie of cookies) {
+        const [key, value] = cookie.split('=');
+            if (!keys || keys.includes(key)) {
+            results[key] = value;
+        }
+    }
+    return results;
+    };
+
+    // Parse cookies from document.cookie
+    const cookies = parseCookies();
+
+    // Filter cookies by keys if provided
+    return filterCookies(cookies, keys);
```

Comment:
Great work with this function.  I think this can be a _bit_ more clear

```suggestion
    const parsedCookies: string[] = parseCookies();

    // Filter cookies by keys if provided
    return filterCookies(parsedCookies, keys);
```